orttraining/orttraining/core/graph/pipeline_transformer.cc

```diff
@@ -825,6 +819,206 @@ void AddNewScalarNodeArgAndInitializer(Graph& graph,
   graph.AddInitializedTensor(proto_data);
 }
 
+Status FindAllConnectedNodes(Graph& graph,
```

Comment:
If this is a new function, could you add a comment on the purpose -- e.g., what a "connected node" is in this case, and any assumptions on the parameters (e.g., should the caller pass in empty collections).

Reply:
Agree.  Looks like this function finds `graph inputs`, `intermediate graph nodes`, and `graph outputs` connected to `const Node* node`. The three fields form an `envelop` of `const Node* node`.

---
In reply to: [452252240](https://github.com/microsoft/onnxruntime/pull/4321#discussion_r452252240) [](ancestors = 452252240)

Reply:
Added a description about this function. 

---
In reply to: [452385293](https://github.com/microsoft/onnxruntime/pull/4321#discussion_r452385293) [](ancestors = 452385293,452252240)